### PR TITLE
[IA-549]  Create an interface that given an org unit allows to make a selection of its children on a map

### PR DESCRIPTION
**Pull request by kaiobrito**

### CODE REVIEW

**Comment by beygorghor** [comment]
Why not using https://react-leaflet.js.org/ ?

**Comment by kaiobrito** [comment]
@beygorghor We are using it!

**Comment by beygorghor** [comment]
Yes just saw it right now sorry !

File: plugins/polio/tests.py [hunk 1]
```diff
@@ -147,6 +147,27 @@ def test_refresh_preparedness_data(self):
...
+                "round_two": {},
+                "obr_name": "campaign with org units",
+                "group": {"name": "hidden group", "org_units": list(map(lambda org_unit: org_unit.id, self.org_units))},
+            },
+            format="json",
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        campaign.refresh_from_db()
+        self.assertEqual(campaign.group.org_units.count(), 2)
```
Comment (xavierbarbosa) [inline plugins/polio/tests.py:169]: nice to have the test to the edge case 👍

File: plugins/polio/tests.py [hunk 2]
```diff
@@ -177,7 +198,7 @@ def test_can_create_and_update_campaign_with_orgunits_group(self):
                 "round_one": {},
                 "round_two": {},
                 "obr_name": "campaign with org units",
-                "group": {"name": "hidden group", "org_units": map(lambda org_unit: org_unit.id, self.org_units)},
+                "group": {"name": "hidden group", "org_units": list(map(lambda org_unit: org_unit.id, self.org_units))},
```
Comment (xavierbarbosa) [inline plugins/polio/tests.py:201]: won't `map` already return a list?

Comment (kaiobrito) [reply plugins/polio/tests.py:201]: It returns a map object and to convert it to an array we have to add list call. There are small differences on map objects and arrays/lists, e.g: map object is not subscriptable so `map(...)[0]` doesn't work. I just wanted to make sure the tests are matching the data coming from the front end.

File: iaso/models/base.py [hunk 3]
```diff
@@ -421,6 +426,7 @@ class Group(models.Model):
     updated_at = models.DateTimeField(auto_now=True)
 
     objects = DefaultGroupManager()
+    domain_objects = DomainGroupManager()
```
Comment (xavierbarbosa) [inline iaso/models/base.py:429]: Nice solution 👍

**Comment by madewulf** [comment]
Ok, I finally got this to work. What was troubling to me is that if you pick as initial org unit an org unit that does not have any sub regions that have shapes, no map will be drawn at all. There should be at least a message saying "no shapes available" or something like that to help understand that. 

Outside of that, I have a bounding box problem. The map is zooming for me 1 level too close to the shape, I would say, so that the shapes are actually not all shown. 
It looks like this: 
Screenshot 2021-06-20 at 10 03 37

While it should look like that: 
Screenshot 2021-06-20 at 09 59 02

Outside of that the feature is starting to be good, with selection and saving working very well, but you missed two points of the request:

- you should take the country as the org unit for which we consider the sub regions (so, find the first org unit parent of the initial_org_unit that is of type country. Set the type of the country as an environment variable, to help here, and notice that just loading the json of the initial_org_unit will give you the list of all parents, so it's easy to find there the country, if you do have the org_unit_type_id of a country). Another option that feels ok to me is to just add a "country" field to the campaign and allow to only pick among the countries there, then use that field as the top parent for map selection afterwards. 

- we don't always want the sub regions to be direct children of the country, actually, there is usually one level (the province)  between the country and the districts (which is the most common name for the type of org unit that we are interested in). So, to load the shapes to show on the map you should load: all org units that have shapes, are children (or grandchildren, or grandgrandchildren, etc ...)  of the country, and that are of type district (again, we can setup the org unit type  id of the district as an environment variable).  

Finally, this component should end up in the bluesquare component library, so that it can be reused elsewhere, but it's ok for now to finish this PR before moving that component to the library, which can be made in another PR.

**Comment by kaiobrito** [comment]
@madewulf it should be good now
